Deprecate kubevirt_vmi_migration_data_total_bytes

sradco opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as a draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
#### Before this PR:
kubevirt_vmi_migration_data_total_bytes doesnt comply with the naming convention and fails the metrics name linter.

#### After this PR:
We deprecated kubevirt_vmi_migration_data_total_bytes in favor of kubevirt_vmi_migration_data_bytes_total, in order to comply with the metrics naming conventions.

### References
- Fixes #https://issues.redhat.com/browse/CNV-71409
<!-- optional,
  Use `Fixes #<issue number>(, Fixes #<issue_number>, ...)` format, to close the issue(s) when PR gets merged.
  Use `Partially addresses #<issue number>` to link an issue without closing it when the PR merges.
  

- Partially addresses #
-->
<!-- optional,
  VEP tracking issue if this PR is implementing one.
  For additional info about VEP tracking issue, see https://github.com/kubevirt/enhancements#process
  
- VEP tracking issue: https://github.com/kubevirt/enhancements/issues/<vep_tracking_issue_number>
-->

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least one e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
kubevirt_vmi_migration_data_total_bytes is deprecated in favor of kubevirt_vmi_migration_data_bytes_total, in order to comply with the metrics naming conventions.
```

sradco commented:
/retest

sradco commented on docs/observability/metrics.md:
```diff
@@ -207,6 +207,9 @@ The total Guest OS data processed and migrated to the new VM. Type: Gauge.
+### kubevirt_vmi_migration_data_total_bytes
+[Deprecated] Replaced by kubevirt_vmi_migration_data_bytes_total. Type: Counter.
+
 ### kubevirt_vmi_migration_data_total_bytes
```
No, Its the old name with the deprecation notice.

avlitman replied:
I talked about the second one, there is 2  kubevirt_vmi_migration_data_total_bytes one should be  kubevirt_vmi_migration_data_bytes_total

sradco commented on pkg/monitoring/rules/recordingrules/vmi.go:
```diff
@@ -57,4 +57,12 @@ var vmiRecordingRules = []operatorrules.RecordingRule{
+		},
+		MetricType: operatormetrics.CounterType,
+		Expr:       intstr.FromString("kubevirt_vmi_migration_data_bytes_total"),
+	},
```
We dont add StabilityLevel field. We need tp remove this from the doc-generator.go

enp0s3 reviewed (commented):
/approve

enp0s3 commented:
/hold
@sradco Please unhold upon rebase

sradco committed:
Deprecated kubevirt_vmi_migration_data_total_bytes

We deprecated kubevirt_vmi_migration_data_total_bytes
in favor of kubevirt_vmi_migration_data_bytes_total,
in ordet to comply with the metrics naming conventions.

Signed-off-by: Shirly Radco <sradco@redhat.com>

sradco commented:
/unhold

sradco commented:
/test pull-kubevirt-build-arm64
/test pull-kubevirt-e2e-k8s-1.34-sig-performance
